website/docs/developer/research/research-lambda-container.md

```diff
@@ -148,6 +148,152 @@ Based on this research, it looks like using lambda as containers is possible whe
 
 If the above issues are solved, then implementing lambda as container is recommended. One way to use private ECR repo is to use a script or terraform code if possible that will create the ECR repo and then build, tag and push the container image to that repo. One that is done, update the terraform lambda modules and deploy the lambda. An additional [card](https://bugs.earthdata.nasa.gov/browse/ORCA-375) has been created to look into this way.
 
+
+### Modifying get_current_archive_list and perform_orca_reconcile
+
+The Orca Internal Reconciliation workflow lambdas require an alternative approach.
+- The maximum time limit for lambdas is 15 minutes. These lambdas may take a significant amount of time, and should not be subject to this limitation.
+
+- Code changes
+  - Merge `get_current_archive_list` and `perform_orca_reconcile` into one codebase.
+  - Wrap functionality in a loop that will process the internal-report queue until no entries remain.
+  - Since ECS does not support timeout, create an overarching timing mechanic that exits if an infinite loop occurs while processing a queue entry.
```

Comment:
could be a problem that it does not support timeout.

Reply:
Yeah, that one seems weird to me. Not sure why they don't do that.

Reply:
So, I feel like you are looking at this through the view of lambda and not ECS/Docker. In this particular case, this is not really an issue. In the case of the queue, once the queue is empty, the task would stop. In the case of a long running item, other timeouts to cause a failure or force the task to stop would be possible like the database connection timing out. So no real need for an "overarching timing mechanic".

Reply:
Our code within the container can have a bug that causes an infinite loop.

Reply:
If that is true, it should be discovered via unit testing and/or validation testing and can be shutdown external to the task so it can be fixed.